### PR TITLE
dev, 3.0: Add tidb-operator in architecture, overview and contribute page, adjust tidb-in-kubernetes TOC

### PR DESCRIPTION
**Pull request by aylei**
Signed-off-by: Aylei <rayingecho@gmail.com>

### What is changed, added or deleted? 
Add tidb-operator in architecture, overview and contribute page

Preview link (with images): https://github.com/aylei/docs-cn/blob/operator-design/dev/reference/tidb-operator-overview.md

### Which version does your change affect? 
`dev` `3.0`

### Checklist 

- [x] No Tab spaces anywhere 
- [x] Leave a blank line both before and after a code block
- [x] Keep the first level heading consistent with `title` in metadata
- [x] Use *four* spaces for each level of indentation except that in `TOC.md`

### CODE REVIEW

**Comment by aylei** [comment]
@lilin90 @dcalvin PTAL
@tennix @weekface @xiaojingchen @DanielZhangQD Please take a technical review

**Comment by aylei** [comment]
Some documents are still in progress, DNM this PR now

File: dev/reference/tidb-operator-overview.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+当集群出现问题需要进行诊断时，你可以：
+
++ 查阅[Kubernetes 上的 TiDB FAQ](faq/tidb-in-kubernetes.md) 寻找是否存在现成的解决办法；
++ 参考[Kubernetes 上的 TiDB 故障诊断](how-to/troubleshoot/tidb-in-kubernetes.md)解决故障。
+
+Kubernetes 上的 TiDB 提供了专用的命令行工具 `tkctl` 用于集群管理和辅助诊断，同时，在 Kubernetes 上，TiDB 的部分生态工具的使用方法也有所不同，你可以：
+
++ 参考[tkctl 使用指南](reference/tools/tkctl.md) 来使用 `tkctl`；
++ 参考[Kubernetes 上的 TiDB 相关工具使用](#未提交) 来了解 TiDB 生态工具在 Kubernetes 上的使用方法。
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
+ 参考 [Kubernetes 上的 TiDB 相关工具使用](#未提交) 来了解 TiDB 生态工具在 Kubernetes 上的使用方法。
```

File: dev/reference/tidb-operator-overview.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
++ [TiDB 集群日志收集](how-to/maintain/tidb-in-kubernetes/log-collecting.md)
+
+当集群出现问题需要进行诊断时，你可以：
+
++ 查阅[Kubernetes 上的 TiDB FAQ](faq/tidb-in-kubernetes.md) 寻找是否存在现成的解决办法；
++ 参考[Kubernetes 上的 TiDB 故障诊断](how-to/troubleshoot/tidb-in-kubernetes.md)解决故障。
+
+Kubernetes 上的 TiDB 提供了专用的命令行工具 `tkctl` 用于集群管理和辅助诊断，同时，在 Kubernetes 上，TiDB 的部分生态工具的使用方法也有所不同，你可以：
+
++ 参考[tkctl 使用指南](reference/tools/tkctl.md) 来使用 `tkctl`；
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
+ 参考 [`tkctl` 使用指南](reference/tools/tkctl.md) 来使用 `tkctl`；
```

File: dev/reference/tidb-operator-overview.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
++ [TiDB 集群升级](/how-to/upgrade/tidb-in-kubernetes.md#升级-tidb-版本)
++ [TiDB 集群配置变更](/how-to/upgrade/tidb-in-kubernetes.md#更新-tidb-集群配置)
++ [TiDB 集群备份恢复](#未merge)
++ [配置 TiDB 集群故障自动转移](how-to/maintain/tidb-in-kubernetes/auto-failover.md)
++ [监控 TiDB 集群](how-to/monitor/tidb-in-kubernetes.md)
++ [TiDB 集群日志收集](how-to/maintain/tidb-in-kubernetes/log-collecting.md)
+
+当集群出现问题需要进行诊断时，你可以：
+
++ 查阅[Kubernetes 上的 TiDB FAQ](faq/tidb-in-kubernetes.md) 寻找是否存在现成的解决办法；
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
+ 查阅 [Kubernetes 上的 TiDB FAQ](faq/tidb-in-kubernetes.md) 寻找是否存在现成的解决办法；
```

File: dev/reference/tidb-operator-overview.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
++ 参考[Kubernetes 上的 TiDB 故障诊断](how-to/troubleshoot/tidb-in-kubernetes.md)解决故障。
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
+ 参考 [Kubernetes 上的 TiDB 故障诊断](how-to/troubleshoot/tidb-in-kubernetes.md)解决故障。
```

File: dev/TOC.md [hunk 5]
```diff
@@ -31,22 +31,22 @@
     - [TiSpark 教程](how-to/get-started/tispark.md)
   + 部署
     - [软硬件环境需求](how-to/deploy/hardware-recommendations.md)
-    - [Kubernetes 上的 TiDB 集群环境需求](how-to/deploy/tidb-in-kubernetes-prerequisites.md)
```
[the remark this answers is not in the document]
Comment (aylei) [reply dev/TOC.md]: Moved to the following section

**Comment by xiaojingchen** [comment]
@aylei  the PR https://github.com/pingcap/docs-cn/pull/1636 has merged, you need to make some adjustments in the overview page.

the how-to/maintain/tidb-in-kubernetes/tidb-cluster.md has been delete and other some pages have been added

**Comment by aylei** [comment]
@xiaojingchen Thanks!
@dcalvin I've addressed the review comments and apply the change to `3.0` PTAL again

File: dev/reference/tidb-operator-overview.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+在任何环境上部署前，都可以参考 [TiDB 集群配置](reference/configuration/tidb-in-kubernetes/cluster-configuration.md)来自定义 TiDB 配置。
+
+部署完成后，你可以参考下面的文档进行 Kubernetes 上 TiDB 集群的使用和运维：
+
++ [管理 TiDB 集群](how-to/maintain/tidb-in-kubernetes/tidb-cluster.md)
++ [访问 TiDB 集群](/how-to/deploy/tidb-in-kubernetes/access-tidb.md)
++ [TiDB 集群扩缩容](/how-to/scale/tidb-in-kubernetes.md)
++ [TiDB 集群升级](/how-to/upgrade/tidb-in-kubernetes.md#升级-tidb-版本)
++ [TiDB 集群配置变更](/how-to/upgrade/tidb-in-kubernetes.md#更新-tidb-集群配置)
++ [TiDB 集群备份恢复](#未merge)
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
+ [TiDB 集群备份恢复](/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md)
```

File: dev/reference/tidb-operator-overview.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+# TiDB Operator 简介
+
+TiDB Operator 是 Kubernetes 上的 TiDB 集群自动运维系统，提供包括部署、升级、扩缩容、备份恢复、配置变更的 TiDB 全生命周期管理。借助 TiDB Operator，TiDB 可以无缝运行在公有云或私有部署的 Kubernetes 集群上。
+
+## TiDB Operator 整体架构
+
+![TiDB Operator Overview](/media/tidb-operator-overview.png)
+
+TiDB Operator 包含自定义资源 `TidbCluster`、`tidb-controller-manager` 组件、`tidb-scheduler` 组件、以及 `tkctl` 命令行工具：
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
TiDB Operator 包含以下组件：
```

File: dev/reference/tidb-operator-overview.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+![TiDB Operator Control Flow](/media/tidb-operator-control-flow.png)
+
+上图是 TiDB Operator 的控制流程解析。由于 TiDB 集群还需要监控、初始化、定时备份、Binlog 等组件，TiDB Operator 中使用 Helm Chart 封装了 TiDB 集群定义。整体的控制流程如下：
+
+1. 用户通过 Helm 创建 `TidbCluster` 对象和相应的一系列 Kubernetes 原生对象，比如执行定时备份的 `CronJob`；
+2. TiDB Operator 会 watch `TidbCluster` 以及其它相关对象，基于集群的实际状态不断调整 PD、TiKV、TiDB 的 `StatefulSet` 和 `Service` 对象；
+3. Kubernetes 的原生控制器根据 `StatefulSet`、`Deployment`、`CronJob` 等对象创建更新或删除对应的 `Pod`；
+4. PD、TiKV、TiDB 的 `Pod` 声明中会指定使用 `tidb-scheduler` 调度器，`tidb-scheduler` 会在调度对应 `Pod` 时应用 TiDB 的特定调度逻辑。
+
+基于上述的声明式控制流程，TiDB Operator 能够自动进行集群节点健康检查和故障恢复，部署、升级、扩缩容等操作也可以通过修改 `TidbCluster` 对象声明“一键”完成。
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
基于上述的声明式控制流程，TiDB Operator 能够自动进行集群节点健康检查和故障恢复。部署、升级、扩缩容等操作也可以通过修改 `TidbCluster` 对象声明“一键”完成。
```

File: dev/reference/tidb-operator-overview.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+当集群出现问题需要进行诊断时，你可以：
+
++ 查阅 [Kubernetes 上的 TiDB FAQ](faq/tidb-in-kubernetes.md) 寻找是否存在现成的解决办法；
++ 参考 [Kubernetes 上的 TiDB 故障诊断](how-to/troubleshoot/tidb-in-kubernetes.md)解决故障。
+
+Kubernetes 上的 TiDB 提供了专用的命令行工具 `tkctl` 用于集群管理和辅助诊断，同时，在 Kubernetes 上，TiDB 的部分生态工具的使用方法也有所不同，你可以：
+
++ 参考 [`tkctl` 使用指南](reference/tools/tkctl.md) 来使用 `tkctl`；
++ 参考 [Kubernetes 上的 TiDB 相关工具使用](#未提交) 来了解 TiDB 生态工具在 Kubernetes 上的使用方法。
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
+ 参考 [Kubernetes 上的 TiDB 相关工具使用](/reference/tools/tools-in-kubernetes.md) 来了解 TiDB 生态工具在 Kubernetes 上的使用方法。
```

**Comment by aylei** [comment]
@dcalvin All comments are addressed, PTAL again

**Pull request merged**